app/models/task.rb

```diff
@@ -0,0 +1,71 @@
+# == Schema Information
+#
+# Table name: tasks
+#
+#  activity_message_value    :string(65536)    not null
+#  id                        :bigint(8)        not null, primary key
+#  job_class                 :string(255)      not null
+#  job_enqueued_for_retry_at :datetime
+#  job_first_started_at      :datetime
+#  job_last_failed_at        :datetime
+#  job_succeeded_at          :datetime
+#  job_type_value            :string(255)      not null
+#  num_attempts              :integer          default(0), not null
+#  provider_job_id           :string
+#
+
+class Task < ApplicationRecord
+  def enqueue(job_params: {})
+    job = job_class.constantize.perform_later(job_params.merge(task_id: id))
+    self.update_attribute(:provider_job_id, job.provider_job_id)
+  end
+
+  def status
+    if pending?
+      :pending
+    elsif in_progress?
+      :in_progress
+    elsif succeeded?
+      :succeeded
+    elsif stalled?
+      :stalled
+    end
+  end
+
+  def start
```

Comment:
is there a better way to approach naming these 'lifecycle' methods?

Reply:
I don't have one at the moment, we can always refactor if it's not working out.